runtime/onert/core/include/ir/train/TrainableGraph.h

```diff
@@ -127,6 +127,11 @@ class TrainableGraph : public IGraph
 public:
   const ITrainableOperation &operation(OperationIndex index) const;
 
+private:
+  void assertValidTopologicalOrder(const std::vector<ir::OperationIndex> &order,
+                                   const std::string &order_type) const;
+  void assertValidBackwardTopologicalOrder(std::vector<ir::OperationIndex> order) const;
```

Comment:
From the name of `assert...`, are these for test only? If so, what about moving this into test code?

Reply:
I intended ONERT to throw error upon finding an invalid topological order.

Adding them as unittest instead would only catch some specific pre-determined cases and compile could still end up using some invalid topological order.

Reply:
IMHO, I guess finding backwardTopologicalOrder should always return correct answer. It is well-known problem. We already have several ones from compiler frontend in this repo, and internal repo. It would be better to fix the problem if we've found some bug than additional validate function in release code.